storage/range_data_iter.go

```diff
@@ -44,11 +44,15 @@ type rangeDataIterator struct {
 func newRangeDataIterator(r *Range, e engine.Engine) *rangeDataIterator {
 	r.RLock()
 	startKey := r.Desc().StartKey
-	if startKey.Equal(engine.KeyMin) {
-		startKey = engine.KeyLocalMax
-	}
 	endKey := r.Desc().EndKey
 	r.RUnlock()
+	// The first range in the keyspace starts at KeyMin, which includes the system-local
+	// space. We need the original StartKey to find the range metadata, but the
+	// actual data starts at KeySystemMax.
+	dataStartKey := startKey
+	if startKey.Equal(engine.KeyMin) {
+		dataStartKey = engine.KeySystemMax
```

Comment:
Why are you using KeySystemMax? The data for a range can definitely include system keys! This should be KeyLocalMax.